serveradmin/powerdns/tests.py

```diff
@@ -0,0 +1,697 @@
+
+        When removing a value from a mapped multi attribute of a record object
+        the existing records are removed.
+        """
+        pass
+
+    def test_object_update_attribute_add_for_single_attribute(self):
+        """Test setting a attribute value creates new record(s)
+
+        When setting a initial value for a mapped single attribute of any
+        object of any servertype with a related record using the `records`
+        attribute the necessary records must be created.
+        """
+        pass
+
+    def test_object_update_attribute_change_for_single_attribute(self):
+        """Test changing a attribute value updates existing records
+
+        When changing the value for a mapped single attribute of any object
+        of any servertype with a related record using the `records` attribute
+        the necessary records must be updated.
+        """
+        pass
+
+    def test_object_update_attribute_remove_for_single_attribute(self):
+        """Test removing the attribute value removes existing records
+
+        When removing the value for a mapped single attribute to any object
+        of any servertype with a related record using the `records` attribute
+        the necessary records must be deleted.
+        """
+        pass
+
+    def test_object_update_attribute_add_for_multi_attribute(self):
+        """Test adding a attribute value creates new record(s)
+
+        When adding a value for a mapped multi attribute of any object of any
+        servertype with a related record using the `records` attribute the
+        necessary records must be created.
+        """
+        pass
+
+    def test_object_update_attribute_remove_for_multi_attribute(self):
+        """Test removing a attribute value removes record(s)
+
+        When removing a value for a mapped multi attribute of any object of
+        any servertype with a related record using the `records` attribute the
+        matching records must be removed.
+        """
+        pass
+
+
+class PowerDNSRecordDeletionTests(PowerDNSTests):
+    """Test cases for PowerDNS record deletion
+
+    This class covers test cases to ensure the desired behaviour for PowerDNS
+    records deleted by Serveradmin.
+    """
+
+    def test_object_deletion_removes_records(self):
+        """Test deletion of a object removes matching records
+
+        When deleting any object of any servertype with `records` attribute
+        set, records related to this object must be removed.
+        """
+
+        domain_query = _create_object('domain', ns=self.ns)
+        record_query = _create_object('record',
+                                      domain=domain_query.get()['hostname'])
+        lb_query = _create_object('loadbalancer', intern_ip='10.0.0.1',
+                                  records=[record_query.get()['hostname']])
+
+        object_id = lb_query.get()['object_id']
+
+        # The A record for the intern_ip should exist at least
+        self.assertGreater(
```

Comment:
Nice, exactly like that for above things. :)

Reply:
I will try to use this more.

Reply:
It's not needed in general for tests, but with modifiable/"dynamic" fixtures I think it's good to check first before making the action to make the test more meaningful.